src/sources/soundsourcepluginlibrary.cpp

```diff
@@ -86,7 +86,7 @@ bool SoundSourcePluginLibrary::init() {
     }
 }
 
-SoundSourceProviderPointer SoundSourcePluginLibrary::createSoundSourceProvider() const {
+SoundSourceProviderPointer SoundSourcePluginLibrary::getSoundSourceProvider() const {
```

Comment:
My last commit internally caches a singular instance of the 
SoundSourceProvider. Now the name of the function matches the 
implementation. But the implementation shouldn't matter to the caller 
anyway, as long as the functions does what its name promises. And the caller 
simply asks for a SoundSourceProvider and does not care if a new one is 
created especially for him or if a shared instance is returned. That's the 
reason why the name "get" is appropriate here. The member function does not 
alter the internal state (const) and returns something that the caller asked 
for.

SoundSourceProvider is a simple factory interface and all our 
implementations are either stateless or immutable so sharing a single 
instance between threads is safe. I added a comment in the documentation of 
the interface for future implementations.

This new implementation is even better, because it fails fast while loading 
the library if no SoundSourceProvider could be created. The subsequent error 
handling in the provider registry has become obsolete. But I left a 
DEBUG_ASSERT in there, just in case ;)

On 21.08.2016 00:46, Daniel Schürmann wrote:
>
> In src/sources/soundsourcepluginlibrary.cpp 
> <https://github.com/mixxxdj/mixxx/pull/994#discussion_r75585954>:
>
> > @@ -86,7 +86,7 @@ bool SoundSourcePluginLibrary::init() {
> >      }
> >  }
> >
> > -SoundSourceProviderPointer SoundSourcePluginLibrary::createSoundSourceProvider() const {
> > +SoundSourceProviderPointer SoundSourcePluginLibrary::getSoundSourceProvider() const {
>
> The old name was not that bad. What was the reason for the change?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/994/files/802196c542a757763ccca53c6f9b6adfac00e1f1#r75585954>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-ag2bzJHJ4rr8CoD65OQJSJwlD-aVks5qh4PfgaJpZM4Jo7bC>.
>